src/sage/schemes/elliptic_curves/mod_poly.py

```diff
@@ -0,0 +1,130 @@
+r"""
+Modular polynomials for elliptic curves
+
+For a positive integer `\ell`, the classical modular polynomial
+`\Phi_\ell \in \ZZ[X,Y]` is characterized by the property that its
+zero set is exactly the set of pairs of `j`-invariants connected
+by a cyclic `\ell`-isogeny.
+
+AUTHORS:
+
+- Lorenz Panny (2023)
+"""
+
+from sage.structure.element import parent
+
+from sage.rings.integer_ring import ZZ
+
+from sage.libs.pari import pari
+from cypari2.handle_error import PariError
+
+from sage.databases.db_modular_polynomials import ClassicalModularPolynomialDatabase
+_db = ClassicalModularPolynomialDatabase()
+
+_cache = dict()
+
+def classical_modular_polynomial(l, j=None):
+    r"""
+    Return the classical modular polynomial `\Phi_\ell`, either as a
+    "generic" bivariate polynomial over `\ZZ`, or as an "instantiated"
+    modular polynomial where one variable has been replaced by the
+    given `j`-invariant.
+
+    Generic polynomials are cached up to a certain size of `\ell`,
+    which significantly accelerates subsequent invocations with the
+    same `\ell`. The default bound is `\ell \leq 150`, which can be
+    adjusted by setting ``classical_modular_polynomial.cache_bound``
+    to a different value. Beware that modular polynomials are very
+    large and the amount of memory consumed by the cache will grow
+    rapidly when the bound is set to a large value.
+
+    INPUT:
+
+    - ``l`` -- positive integer.
+    - ``j`` -- either ``None`` or a ring element:
+      * if ``None`` is given, the original modular polynomial
```

Comment:
```suggestion
    - ``j`` -- either ``None`` or a ring element:

      * if ``None`` is given, the original modular polynomial
```
This is needed for the doc to compile.